src/pages/config.tsx

```diff
@@ -136,34 +137,67 @@ function ConfigPage (): React.JSX.Element | null {
     setResetKey((prev) => prev + 1)
   }, [])
 
-  const newlineStringSave = (value: string): string => JSON.stringify(convertUrlInputToArray(value))
-  const newlineStringLoad = (value: string): string => convertUrlArrayToInput(JSON.parse(value))
+  const newlineToJsonArray = (value: string): string => JSON.stringify(convertUrlInputToArray(value))
+  const jsonToNewlineString = (value: string): string => convertUrlArrayToInput(JSON.parse(value))
 
   return (
     <>
     {!isLoadedInIframe && <Header /> }
     <main className='e2e-config-page pa4-l bg-snow mw7 center pa4'>
       <Collapsible collapsedLabel="View config" expandedLabel='Hide config' collapsed={isLoadedInIframe}>
+      <LocalStorageToggle
+        className="e2e-config-page-input"
+        label="Enable Recursive Gateways"
+        description="Use recursive gateways configured below for retrieval of content."
+        defaultValue={defaultEnableRecursiveGateways}
+        localStorageKey={LOCAL_STORAGE_KEYS.config.enableRecursiveGateways}
+        resetKey={resetKey}
+      />
```

Comment:
```suggestion
	      <LocalStorageToggle
	        className="e2e-config-page-input"
	        label="Enable Trustless Gateways"
	        description="Use recursive trustless gateways configured below for retrieval of content."
	        defaultValue={defaultEnableRecursiveGateways}
	        localStorageKey={LOCAL_STORAGE_KEYS.config.enableRecursiveGateways}
	        resetKey={resetKey}
	      />
```

Reply:
There are two things in this suggestion that may not be clear:

1. indentation
2. the change of the label.

indentation should be fixed for sure.

as for the label, how do we have trustless gateway providers if they're named "recursive gateways"?

if we call a trustless gateway for content, we don't control whether they are recursing to retrieve that content or not.. right? I'm slightly confused by the distinction between recursive and trustless gateways with these updates.

Reply:
That's fair, maybe we need to work more on the copy here. 


To start off: all gateways used by verified fetch are trustless. 

Recursive gateways are more expensive to run because they handle retrieval of blocks they don't have. 

Non recursive gateways are typically providers announcing to the IPNI, who serve blocks over http using the trustless gateway. It's typically a cheaper alternative to making the blocks available over bitswap. It's pretty common for Filecoin providers. (The reason that some opt for bitswap is because kubo doesn't support http retrievals yet)

Reply:
My mental model is this:
- Recursive gateways are the "last resort", a fallback. 
  - Ideally, we would have ability to delay hitting them, and try direct retrieval first. 
  - SW comes with a list of those fallback gateways. In our case that will only be `https://trustless-gateway.link`.
- SW may discover other HTTP gateways via routing system 
  - e.g. `/dns4/dag.w3s.link/tcp/443/https` [here](https://delegated-ipfs.dev/routing/v1/providers/bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi)
  - if we discover HTTP gateway in relation to a specific CID
    - it should NOT be added to the list of recursive (fallback) gateways
    - should only used as provider of that specific CID (or DAG, if we keep session info around).
- UI-wise, we want to 
  - use recursive gateways only as a last resort fallback, when direct retrieval from actual provider fails or is not possible, 
  - and have toggle to disable that fallback, so we can see when direct retrieval fails.
  - avoid generic terms like "trustless gateway", just like we would not use "TCP socket address" without any context about the function of the socket :)

To avoid confusion, perhaps we could call things what they are, in order: 
- First, have section `Direct Retrieval` with toggles supported 
  - `Support WebSockets`
  - `Support WebTransport`
  - `Support HTTPS` that controls if we act on gateways discovered via routing system
- Then, at bottom have `Fallback Retrieval` section
  - `Recursive HTTPS Gateways` list
    - And a toggle next to the list to `Use as fallback if direct/p2p retrieval is not possible`
  - By the fact its at the end, it conveys it is a last resort fallback, and not the golden path

Reply:
config sections done in [`7e7ef6f` (#391)](https://github.com/ipfs/service-worker-gateway/pull/391/commits/7e7ef6f8db4098fa4d6144d7e34ce88a1f0f53d2)